commit 26c2c06c83ef73bb51400aef0e3714faeff91fed
Author: Anton Novikov <an146@ya.ru>
Date:   Mon May 16 16:27:58 2016 +0000

    NO_UTMP in sessreg

diff --git a/app/sessreg-1.0.7/sessreg.h b/app/sessreg-1.0.7/sessreg.h
index a5e7f83..a32cf58 100644
--- a/app/sessreg-1.0.7/sessreg.h
+++ b/app/sessreg-1.0.7/sessreg.h
@@ -82,6 +82,13 @@
 #endif
 #endif
 
+#ifndef NO_UTMP
+#define NO_UTMP
+#endif
+#ifdef USE_UTMPX
+#undef USE_UTMPX
+#endif
+
 #ifndef WTMP_FILE
 # ifdef _PATH_WTMP
 #  define WTMP_FILE	_PATH_WTMP
