commit a0fed1108499907c23b8bccfce387d4b4e04647c
Author: Anton Novikov <an146@ya.ru>
Date:   Mon May 16 16:18:59 2016 +0000

    ignore man

diff --git a/app/sessreg-1.0.7/man/Makefile.am b/app/sessreg-1.0.7/man/Makefile.am
index 5ef6009..f499b90 100644
--- a/app/sessreg-1.0.7/man/Makefile.am
+++ b/app/sessreg-1.0.7/man/Makefile.am
@@ -19,4 +19,4 @@ MAN_SUBSTS += -f filenames.sed
 sessreg.$(APP_MAN_SUFFIX): filenames.sed
 
 .man.$(APP_MAN_SUFFIX):
-	$(AM_V_GEN)$(SED) $(MAN_SUBSTS) < $< > $@
+	touch $@
diff --git a/app/sessreg-1.0.7/man/Makefile.in b/app/sessreg-1.0.7/man/Makefile.in
index 2b376db..ebb1bb0 100644
--- a/app/sessreg-1.0.7/man/Makefile.in
+++ b/app/sessreg-1.0.7/man/Makefile.in
@@ -416,7 +416,7 @@ filenames.sed: filenames.sed.c
 sessreg.$(APP_MAN_SUFFIX): filenames.sed
 
 .man.$(APP_MAN_SUFFIX):
-	$(AM_V_GEN)$(SED) $(MAN_SUBSTS) < $< > $@
+	touch $@
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
