--- a/gdbus/mainloop.c
+++ b/gdbus/mainloop.c
@@ -288,7 +288,8 @@
 {
 	DBusConnection *conn;
 
-	conn = dbus_bus_get(type, error);
+	conn = dbus_connection_open( "unix:path=/var/run/bluez/system_bus_socket", error);
+	dbus_bus_register( conn, error);
 
 	if (error != NULL) {
 		if (dbus_error_is_set(error) == TRUE)
@@ -311,7 +311,8 @@
 {
 	DBusConnection *conn;
 
-	conn = dbus_bus_get_private(type, error);
+	conn = dbus_connection_open_private( "unix:path=/var/run/bluez/system_bus_socket", error);
+	dbus_bus_register( conn, error);
 
 	if (error != NULL) {
 		if (dbus_error_is_set(error) == TRUE)
