--- ./README
+++ ../ports/README
@@ -47,10 +47,10 @@
 
 HOW TO BUILD -- UNIX, shared library libbz2.so.
 
-Do 'make -f Makefile-libbz2_so'.  This Makefile seems to work for
-Linux-ELF (RedHat 7.2 on an x86 box), with gcc.  I make no claims
-that it works for any other platform, though I suspect it probably
-will work for most platforms employing both ELF and gcc.
+Do 'make -f Makefile-libbz2_so PREFIX=/xxx/yyy'.  This Makefile seems
+to work for Linux-ELF (RedHat 7.2 on an x86 box), with gcc.  I make 
+no claims that it works for any other platform, though I suspect it 
+probably will work for most platforms employing both ELF and gcc.
 
 bzip2-shared, a client of the shared library, is also built, but not
 self-tested.  So I suggest you also build using the normal Makefile,
