--- a/gio/gdesktopappinfo.c
+++ b/gio/gdesktopappinfo.c
@@ -1258,6 +1258,17 @@ desktop_file_dir_create_for_config (GArray      *array,
   g_array_append_val (array, dir);
 }
 
+/* Axiom add-on function to read desktop files from ~/.applications */
+static void
+desktop_file_dir_create_for_axiom (GArray *array)
+{
+  DesktopFileDir dir = { 0, };
+
+  dir.path = g_build_filename (g_get_home_dir(), ".applications", NULL);
+
+  g_array_append_val (array, dir);
+}
+
 /*< internal >
  * desktop_file_dir_reset:
  * @dir: a #DesktopFileDir
@@ -1467,6 +1478,9 @@ desktop_file_dirs_lock (void)
       for (i = 0; dirs[i]; i++)
         desktop_file_dir_create_for_config (tmp, dirs[i]);
 
+      /* Next, the data from ~/.applications for axiom needs */
+      desktop_file_dir_create_for_axiom (tmp);
+
       /* Now the data.  Highest priority: the user's ~/.local/share/applications */
       desktop_file_dir_user_data_index = tmp->len;
       desktop_file_dir_create (tmp, g_get_user_data_dir ());
