commit 7f13d7d0187f5d04f440fc1a9ad3783125098746
Author: Anton Novikov <an146@ya.ru>
Date:   Tue May 17 10:45:08 2016 +0000

    use termios.h

diff --git a/xorg-server-1.12.2/hw/xfree86/os-support/xf86_OSlib.h b/xorg-server-1.12.2/hw/xfree86/os-support/xf86_OSlib.h
index 9161e18..85e7c83 100644
--- a/xorg-server-1.12.2/hw/xfree86/os-support/xf86_OSlib.h
+++ b/xorg-server-1.12.2/hw/xfree86/os-support/xf86_OSlib.h
@@ -178,7 +178,7 @@
 #include <sys/types.h>
 #include <assert.h>
 
-#ifdef __linux__
+#if 0
 #include <termio.h>
 #else                           /* __GLIBC__ */
 #include <termios.h>
